src/TMS_Geom.h

```
@@ -46,16 +47,76 @@ class TMS_Geom {
     // Set the geometry
     void SetGeometry(TGeoManager *geometry) {
       geom = geometry;
-      std::cout << "Global geometry set to " << geometry->GetName() << std::endl;
       geom->LockGeometry();
+      
+      // There's an overall scale factor depending on if the gmdl was loaded with cm or mm. 
+      // So here we're trying to automatically figure out the scale factor
+      // If the geometry changes too much, then this would not work and so we'd want to give up.
+      TGeoBBox *box = dynamic_cast<TGeoBBox*>(geom->GetTopVolume()->GetShape());
```

Wouldn't this depend on whether we build the full detector hall or some sub-volume? Maybe it is safer to get a part of the TMS detector and compare its size to the known size. Since that might also change a bit, it might be good to put in some reasonable bands of sizes to accept the same scaling factor. But I agree that it is better to throw an error than to do something wrong when we are unsure about the scaling.

---

I was worried that parts would move or change name. The tms groups is talking about doing studies that rotate and remove planes. And there's an upcoming change to the muon window. The g4 sim should always be run with the full geometry to this should be relatively stable.